extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/LateBoundSampler.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.opentelemetry.runtime.tracing;
+
+import java.util.List;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+
+public class LateBoundSampler implements Sampler {
+    private Sampler delegate;
+
+    public void setSamplerDelegate(Sampler delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public SamplingResult shouldSample(Context parentContext,
```

Comment:
Could you add some logging similar to the `BatchSpanProcessor` so that if `delegate` does happen to be null, we're logging a warning?